agentstack/frameworks/langgraph.py

```diff
@@ -0,0 +1,722 @@
+from typing import Optional, Union, Callable, Any
+from dataclasses import dataclass
+from pathlib import Path
+import ast
+from agentstack import conf, log
+from agentstack import packaging
+from agentstack.exceptions import ValidationError
+from agentstack.generation import asttools, InsertionPoint
+from agentstack._tools import ToolConfig
+from agentstack.agents import AgentConfig, get_all_agent_names
+from agentstack.tasks import TaskConfig, get_all_task_names
+from agentstack import graph
+
+ENTRYPOINT: Path = Path('src/graph.py')
+
+GRAPH_NODE_START = 'START'
+GRAPH_NODE_END = 'END'
+GRAPH_NODE_TOOLS = 'tools'  # references the `ToolNode` instance
+GRAPH_NODES_SPECIAL = (GRAPH_NODE_START, GRAPH_NODE_END, )
+
+
+@dataclass
+class LangGraphProvider:
+    """An LLM provider for the LangGraph framework."""
+
+    class_name: str
+    module_name: str
+    dependency: str
+
+
+PROVIDERS = {
```

Comment:
We could store these in a JSON file instead. I chose the popular providers; should we implement them all?

Reply:
I think this is okay for now! If someone opens an issue for supporting a more obscure model, we can add it :)